File: 1. 단계별로_풀어보기/30~39/30. 투 포인터/1. 3273.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const N = +inputArr[0]
const A = inputArr[1].trim().split(' ').map(Number)
const X = +inputArr[2]

// 우선 정렬한다.
A.sort((a, b) => a - b)

// 왼쪽 끝, 오른쪽 끝을 가리킬 변수
let left = 0
let right = A.length - 1

// 결과 변수
let result = 0

// 왼쪽이 더 작은동안 계속 반복한다.
while (left < right) {
  if (A[left] + A[right] === X) {
    result++
    left++
    right--
  } // ::
  else if (A[left] + A[right] < X) {
    left++
  } // ::
  else {
    right--
  }
}

process.stdout.write(`${result}`)

File: 1. 단계별로_풀어보기/30~39/30. 투 포인터/1. 3273.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N = int(input())
A = list(map(int, input().split()))
X = int(input())

A.sort()

left = 0
right = N - 1

result = 0

while left < right:
  if A[left] + A[right] == X:
    result += 1
    left += 1
    right -= 1
  elif A[left] + A[right] < X:
    left += 1
  else:
    right -= 1
  
output(f"{result}")